src/Symfony/Component/Clock/Clock.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock;
+
+/**
+ * A global clock.
+ *
+ * Defined as enum to be final and non-instantiable.
+ */
+enum Clock
+{
+    public static function now(): \DateTimeImmutable
+    {
+        return self::clock()->now();
+    }
+
+    public static function clock(ClockInterface $newClock = null): ClockInterface
+    {
+        static $clock;
+
+        if (null === $newClock) {
+            return $clock ??= new NativeClock();
+        }
+
+        $previousClock = $clock ??= new NativeClock();
+        $clock = $newClock;
```

Comment:
The assignment is not necessary here, right? As it's set in the next line anyway.

```suggestion
        $previousClock = $clock ?? new NativeClock();
        $clock = $newClock;
```

Reply:
right, fixed thanks